homeassistant/config_entries.py

```diff
@@ -943,6 +947,15 @@ async def async_set_disabled_by(
         ent_reg = entity_registry.async_get(self.hass)
 
         if not entry.disabled_by:
+            # Make sure requirements and dependencies of component are resolved
+            try:
+                await async_prepare_setup_component(
+                    self.hass, entry.domain, self._hass_config
+                )
+            except PrepareSetupError as err:
+                _LOGGER.error(
+                    "Failed to load integration %s: %s", entry.domain, str(err)
+                )
```

Comment:
We should throw an exception here, and catch it in `config_entry_disable` websocket command handling